doc/stdenv.xml

```diff
@@ -179,68 +179,285 @@ genericBuild
+  For now, the run-time/build-time distinction is just a hint for mental clarity, but in the future it perhaps could be enforced.
+</para>
+<para>
+  The extension of <envar>PATH</envar> with dependencies, alluded to above, proceeds according to the relative platforms alone.
+  The process is carried only for dependencies whose host platform matches the new derivation's build platform–i.e. which runs on the directory where the new derivation will be built.
+  <footnote><para>Currently, that means all dependencies are put on the <envar>PATH</envar> for native builds, but in the future to match cross that may not be the case—the platforms would be assumed to be unique for consistency between cross and native builds.</para></footnote>
+  Of those, for each dependency <replaceable>dir</replaceable>, <filename><replaceable>dir</replaceable>/bin</filename>, if present, is added to the <envar>PATH</envar> environment variable.
+</para>
+<para>
+  The dependency is propagated when it forces some of its other-transitive (non-immediate) downstream dependencies to also take it on as an immediate dependency.
+  Nix itself already takes a package's transitive dependencies into account, but this propagation ensures nixpkgs-specific infrastructure like setup hooks (mentioned above) also are run as if the propagated dependency.
+</para>
+<para>
+  It is important to note dependencies are not necessary propagated as the same sort of dependency that they were before, but rather as the corresponding sort so that the platform rules still line up.
+  The exact rules for dependency propagation can be given by assigning each sort of dependency two integers based one how it's host and target platforms are offset from the depending derivation's platforms.
+  Those offsets are given are given below in the descriptions of each dependency list attribute.
+  Informally, we traverse propagated inputs, accumulating an dependency for each transitive path such that:
+  <itemizedlist>
+    <listitem>
+      <para>
+      The type of dependency is determined sum-like fold of the offsets assigned to each edge of the path
+      </para>
+    </listitem>
+    <listitem>
+      <para>
+      The path is only included if the subpaths sharing the starting point never go out of bounds (<literal>&lt; -2</literal>, <literal>&gt; 1</literal>).
+      This is a fancy way of saying we prune as we go.
+      </para>
+    </listitem>
+  </itemizedlist>
+  Formally:
+  <programlisting>
+let mapOffset(h, t, i) = i + (if i <= 0 then h else t - 1)
+
+propagated-dep(h0, t0, A, B)
+propagated-dep(h1, t1, B, C)
+h0 + h1 in {-1, 0, 1}
+h0 + t1 in {-1, 0, 1}
+-------------------------------------- Transitive property
+propagated-dep(mapOffset(h0, t0, h1),
+               mapOffset(h0, t0, t1),
+               A, C)</programlisting>
```

Comment:
Is there precendent for using judgment notation in nix documentation? If not perhaps some background discussion is in order. If so then hopefully there is some friendly introduction that we can cite.

Reply:
haha hell no there are isn't. Know of any off hand?

Reply:
Added wikipedia on natural deduction for now.